### PR TITLE
[IMP] product_variant_cost_price: synchronize cost method when product and template relation is one2one

### PR DESCRIPTION
**Pull request by agaldona**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Include a test for this, please

**Comment by agaldona** [comment]
Tests added :wink:

File: product_variant_cost_price/models/product_product.py [hunk 1]
```diff
@@ -70,6 +70,14 @@ def write(self, values):
         if 'standard_price' in values:
             for product in self:
                 product._set_standard_price(product, values['standard_price'])
+        if (values.get('cost_method', False) and not
+                self.env.context.get('force_not_load', False)):
+            cost_method = values.get('cost_method', False)
+            templates = self.mapped('product_tmpl_id').filtered(
+                lambda x: len(x.product_variant_ids) == 1 and
+                x.cost_method != cost_method)
+            templates.with_context(force_not_load=True).write(
```
Comment (pedrobaeza) [inline product_variant_cost_price/models/product_product.py:79]: Isn't this creating a circular write?

Comment (agaldona) [inline product_variant_cost_price/models/product_product.py:79]: No, I use context value for that problem. `self.env.context.get('force_not_load', False)`

**Comment by pedrobaeza** [comment]
:+1:

**Comment by oihane** [comment]
There aren't test to check what happens if the template has multiple variants.

**Comment by agaldona** [comment]
Tests added for templates with multiple variants  too.

**Comment by oihane** [comment]
Thanks :+1:

**Comment by Daniel-CA** [comment]
:+1:

**Pull request merged**